MediaBrowser.Api/QuickConnect/QuickConnectService.cs

```diff
@@ -0,0 +1,132 @@
+using System;
+using MediaBrowser.Controller.Configuration;
+using MediaBrowser.Controller.Library;
+using MediaBrowser.Controller.Net;
+using MediaBrowser.Controller.QuickConnect;
+using MediaBrowser.Model.QuickConnect;
+using MediaBrowser.Model.Services;
+using Microsoft.Extensions.Logging;
+
+namespace MediaBrowser.Api.QuickConnect
+{
+    [Route("/QuickConnect/Initiate", "GET", Summary = "Requests a new quick connect code")]
+    public class Initiate : IReturn<QuickConnectResult>
+    {
+        [ApiMember(Name = "FriendlyName", Description = "Device friendly name", IsRequired = false, DataType = "string", ParameterType = "query", Verb = "GET")]
+        public string FriendlyName { get; set; }
+    }
+
+    [Route("/QuickConnect/Connect", "GET", Summary = "Attempts to retrieve authentication information")]
+    public class Connect : IReturn<QuickConnectResult>
+    {
+        [ApiMember(Name = "Secret", Description = "Quick connect secret", IsRequired = true, DataType = "string", ParameterType = "query", Verb = "GET")]
+        public string Secret { get; set; }
+    }
+
+    [Route("/QuickConnect/Authorize", "POST", Summary = "Authorizes a pending quick connect request")]
+    [Authenticated]
+    public class Authorize : IReturn<bool>
+    {
+        [ApiMember(Name = "Code", Description = "Quick connect identifying code", IsRequired = true, DataType = "string", ParameterType = "query", Verb = "GET")]
+        public string Code { get; set; }
+    }
+
+    [Route("/QuickConnect/Deauthorize", "POST", Summary = "Deletes all quick connect authorization tokens for the current user")]
+    [Authenticated]
+    public class Deauthorize : IReturn<int>
+    {
+        [ApiMember(Name = "UserId", Description = "User Id", IsRequired = false, DataType = "string", ParameterType = "query", Verb = "GET")]
+        public Guid UserId { get; set; }
+    }
+
+    [Route("/QuickConnect/Status", "GET", Summary = "Gets the current quick connect state")]
+    public class QuickConnectStatus : IReturn<QuickConnectResult>
+    {
+
+    }
+
+    [Route("/QuickConnect/Available", "POST", Summary = "Enables or disables quick connect")]
+    [Authenticated(Roles = "Admin")]
+    public class Available : IReturn<QuickConnectState>
+    {
+        [ApiMember(Name = "Status", Description = "New quick connect status", IsRequired = false, DataType = "QuickConnectState", ParameterType = "query", Verb = "GET")]
+        public QuickConnectState Status { get; set; }
+    }
+
+    [Route("/QuickConnect/Activate", "POST", Summary = "Temporarily activates quick connect for the time period defined in the server configuration")]
+    [Authenticated]
```

Comment:
```suggestion
    [Authenticated(Roles = "Admin")]
```

Why would someone use this endpoint?

Reply:
When quick connect has been enabled by the server administrator (who changes the state from ``unavailable`` -> ``available``), quick connect does not accept any incoming initiation requests until it is temporarily activated by an already logged in user. This is to protect the ``/Initiate`` endpoint from being flooded with malicious requests, which could exhaust all of the possible short codes. After the time limit expires, it transitions from ``active`` -> ``available``.